src/database/models/creator_follows.rs

```diff
@@ -0,0 +1,97 @@
+use itertools::Itertools;
+
+use super::{OrganizationId, UserId};
+use crate::database::models::DatabaseError;
+
+pub struct UserFollow {
+    pub follower_id: UserId,
+    pub target_id: UserId,
+}
+
+pub struct OrganizationFollow {
+    pub follower_id: UserId,
+    pub target_id: OrganizationId,
+}
+
+struct FollowQuery {
+    follower_id: i64,
+    target_id: i64,
+}
+
+impl From<FollowQuery> for UserFollow {
+    fn from(value: FollowQuery) -> Self {
+        UserFollow {
+            follower_id: UserId(value.follower_id),
+            target_id: UserId(value.target_id),
+        }
+    }
+}
+
+impl From<FollowQuery> for OrganizationFollow {
+    fn from(value: FollowQuery) -> Self {
+        OrganizationFollow {
+            follower_id: UserId(value.follower_id),
+            target_id: OrganizationId(value.target_id),
+        }
+    }
+}
+
+macro_rules! impl_follow {
+    ($target_struct:ident, $table_name:tt, $target_id_type:ident, $target_id_ctor:expr) => {
+        impl $target_struct {
+            pub async fn insert(
+                &self,
+                exec: impl sqlx::Executor<'_, Database = sqlx::Postgres>,
+            ) -> Result<(), DatabaseError> {
+                sqlx::query!(
+                    " INSERT INTO " + $table_name + " (follower_id, target_id) VALUES ($1, $2)",
```

Comment:
does sqlx support using the prepared statement for the table name? like INSERT INTO $1? I think it would be worth trying bc I am uneasy abt string concat in sql statements (this is totally safe, but I'm worried it's a bad practice)

Reply:
As far as I can tell, no, unfortunately (https://stackoverflow.com/questions/25631006/postgres-sqlx-prepared-statement-with-table-name-bindvar, links https://www.postgresql.org/docs/9.1/xfunc-sql.html). It's a Postgres limitation. Not a huge fan either, but the only other alternatives I see are duplicating or using dynamic (unchecked) queries, which would still require string concatenation (but not the macro, at least).